src/aria/grid/grid.ts

```diff
@@ -0,0 +1,272 @@
+          (widgetElement !== undefined && cell.element() === widgetElement)
+        ) {
+          return cell;
+        }
+      }
+    }
+    return;
+  }
+}
+
+/** A directive that represents a row in a grid. */
+@Directive({
+  selector: '[ngGridRow]',
+  exportAs: 'ngGridRow',
+  host: {
+    'class': 'grid-row',
+    '[attr.role]': 'role()',
+  },
+})
+export class GridRow {
+  /** A reference to the host element. */
+  private readonly _elementRef = inject(ElementRef);
+
+  /** The cells that make up this row. */
+  private readonly _cells = contentChildren(GridCell);
+
+  /** The UI patterns for the cells in this row. */
+  private readonly _cellPatterns: Signal<GridCellPattern[]> = computed(() =>
+    this._cells().map(c => c.pattern),
+  );
+
+  /** The parent grid. */
+  private readonly _grid = inject(Grid);
+
+  /** The parent grid UI pattern. */
+  readonly grid = computed(() => this._grid.pattern);
+
+  /** The host native element. */
+  readonly element = computed(() => this._elementRef.nativeElement);
+
+  /** The ARIA role for the row. */
+  readonly role = input<'row' | 'rowheader'>('row');
+
+  /** The index of this row within the grid. */
+  readonly rowIndex = input<number>();
+
+  /** The UI pattern for the grid row. */
+  readonly pattern = new GridRowPattern({
+    ...this,
+    cells: this._cellPatterns,
+  });
+}
+
+/** A directive that represents a cell in a grid. */
+@Directive({
+  selector: '[ngGridCell]',
+  exportAs: 'ngGridCell',
+  host: {
+    'class': 'grid-cell',
+    '[attr.role]': 'role()',
+    '[attr.rowspan]': 'pattern.rowSpan()',
+    '[attr.colspan]': 'pattern.colSpan()',
+    '[attr.data-active]': 'pattern.active()',
+    '[attr.aria-disabled]': 'pattern.disabled()',
+    '[attr.aria-rowspan]': 'pattern.rowSpan()',
+    '[attr.aria-colspan]': 'pattern.colSpan()',
+    '[attr.aria-rowindex]': 'pattern.ariaRowIndex()',
+    '[attr.aria-colindex]': 'pattern.ariaColIndex()',
+    '[attr.aria-selected]': 'pattern.ariaSelected()',
+    '[tabindex]': 'pattern.tabIndex()',
+  },
+})
+export class GridCell {
+  /** A reference to the host element. */
+  private readonly _elementRef = inject(ElementRef);
+
+  /** The widget contained within this cell, if any. */
+  private readonly _widget = contentChild(GridCellWidget);
+
+  /** The UI pattern for the widget in this cell. */
+  private readonly _widgetPattern: Signal<GridCellWidgetPattern | undefined> = computed(
+    () => this._widget()?.pattern,
+  );
+
+  /** The parent row. */
+  private readonly _row = inject(GridRow);
+
+  /** A unique identifier for the cell. */
+  private readonly _id = inject(_IdGenerator).getId('ng-grid-cell-');
+
+  /** The host native element. */
+  readonly element = computed(() => this._elementRef.nativeElement);
+
+  /** The ARIA role for the cell. */
+  readonly role = input<'gridcell' | 'columnheader'>('gridcell');
+
+  /** The number of rows the cell should span. */
+  readonly rowSpan = input<number>(1);
```

Comment:
nit: maybe `rowspan`, `colspan`, `rowindex`, and `colindex` to match native html table attributes